src/options/retry.rs

```diff
@@ -0,0 +1,32 @@
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+/// A command retry policy.
+pub struct RetryOptions {
+    pub(crate) limit: usize,
+    pub(crate) delay: std::time::Duration,
+}
+
+impl Default for RetryOptions {
+    fn default() -> Self {
+        Self {
+            limit: 3,
+            delay: std::time::Duration::from_millis(500),
+        }
+    }
+}
+
+impl RetryOptions {
+    /// Sets how many time we retry a failing command before giving up.
+    pub fn retry_limit(self, limit: usize) -> Self {
+        Self { limit, ..self }
+    }
+
+    /// Keep retrying regardless how many time we failed.
+    pub fn retry_endlessly(self) -> Self {
```

Comment:
**SUGGESTION:** Not a blocker, but maybe `retry_forever` like in Polly (https://github.com/App-vNext/Polly#retry-forever-until-succeeds) would be a better name?

Reply:
Done